bundles/framework/maplegend/Flyout.js

```diff
@@ -127,10 +127,27 @@ Oskari.clazz.define('Oskari.mapframework.bundle.maplegend.Flyout',
                 layer = layers[n];
                 layerContainer = this._createLayerContainer(layer);
 
+                const layerTitle = jQuery('<div class="maplegend-layer-title">' + layer.getName() + '</div>');
+                layerTitle.append(me.templateTools.clone());
+
+                const uuid = layer.getMetadataIdentifier();
+
+                if (!uuid) {
+                    layerTitle.find('div.layer-description').hide(); // no functionality -> hide
+                } else {
+                    layerTitle.find('div.icon-info').on('click', function (event) {
+                        event.stopPropagation();
+
+                        sandbox.postRequestByName('catalogue.ShowMetadataRequest', [{
+                            uuid: uuid
+                        }]);
+                    });
+                }
```

Comment:
It's cleaner to add the tool container when needed instead of always adding it and removing it if there's no uuid:
```suggestion
                const uuid = layer.getMetadataIdentifier();
                if (uuid) {
                    layerTitle.append(me.templateTools.clone());
                    layerTitle.find('div.icon-info').on('click', function (event) {
                        event.stopPropagation();
                        sandbox.postRequestByName('catalogue.ShowMetadataRequest', [{
                            uuid: uuid
                        }]);
                    });
                }
```